libs/langchain/langchain/chat_models/databricks.py

```diff
@@ -0,0 +1,195 @@
+
+    To use, you should have the ``mlflow`` python package installed.
+    For more information, see https://mlflow.org/docs/latest/llms/deployments/databricks.html.
+
+    Example:
+        .. code-block:: python
+
+            from langchain.chat_models import ChatDatabricks
+
+            chat = ChatDatabricks(
+                target_uri="<target_uri>",
+                endpoint="<endpoint>",
+                params={
+                    "temperature": 0.1
+                }
+            )
+    """
+
+    endpoint: str
+    target_uri: str = "databricks"
+    params: Optional[ChatParams] = None
+    client: Any = None
+
+    def __init__(self, **kwargs: Any):
+        super().__init__(**kwargs)
+        try:
+            from mlflow.deployments import get_deploy_client
+
+            self.client = get_deploy_client(self.target_uri)
+        except ImportError as e:
+            raise ImportError(
+                "Failed to create the client. "
+                "Please install mlflow with `pip install mlflow`."
+            ) from e
+
+    @property
+    def _default_params(self) -> Dict[str, Any]:
+        params: Dict[str, Any] = {
+            "target_uri": self.target_uri,
+            "endpoint": self.endpoint,
+            **(self.params.dict() if self.params else {}),
+        }
+        return params
+
+    def _generate(
+        self,
+        messages: List[BaseMessage],
+        stop: Optional[List[str]] = None,
+        run_manager: Optional[CallbackManagerForLLMRun] = None,
+        **kwargs: Any,
+    ) -> ChatResult:
+        message_dicts = [
+            ChatDatabricks._convert_message_to_dict(message) for message in messages
+        ]
+        data: Dict[str, Any] = {
+            "messages": message_dicts,
+            **(self.params.dict() if self.params else {}),
+        }
+
+        resp = self.client.predict(endpoint=self.endpoint, inputs=data)
+        return ChatDatabricks._create_chat_result(resp)
+
+    async def _agenerate(
+        self,
+        messages: List[BaseMessage],
+        stop: Optional[List[str]] = None,
+        run_manager: Optional[AsyncCallbackManagerForLLMRun] = None,
+        **kwargs: Any,
+    ) -> ChatResult:
+        func = partial(
+            self._generate, messages, stop=stop, run_manager=run_manager, **kwargs
+        )
+        return await asyncio.get_event_loop().run_in_executor(None, func)
+
+    @property
+    def _identifying_params(self) -> Dict[str, Any]:
+        return self._default_params
+
+    def _get_invocation_params(
+        self, stop: Optional[List[str]] = None, **kwargs: Any
+    ) -> Dict[str, Any]:
+        """Get the parameters used to invoke the model FOR THE CALLBACKS."""
+        return {
+            **self._default_params,
+            **super()._get_invocation_params(stop=stop, **kwargs),
+        }
+
+    @property
+    def _llm_type(self) -> str:
+        """Return type of chat model."""
+        return "databricks-chat"
```

Comment:
ditto. Please also add this new llm type to [get_type_to_cls_dict in llms/__init__.py](https://github.com/langchain-ai/langchain/blob/686162670e2fe15fb906999da84d36a273b2f25e/libs/langchain/langchain/llms/__init__.py#L761) so that it can be correctly serialized/unserialized using LangChain.

Reply:
Added